kasa/tests/smart/modules/test_fan.py

```diff
@@ -48,3 +52,31 @@ async def test_sleep_mode(dev: SmartDevice, mocker: MockerFixture):
 
     assert fan.sleep_mode is True
     assert sleep_feature.value is True
+
+
+@fan
+async def test_fan_interface(dev: SmartDevice, mocker: MockerFixture):
+    """Test fan speed on device interface."""
+    assert isinstance(dev, SmartDevice)
+    fan = cast(FanModule, dev.get_module("FanModule"))
+    device = fan._device
+    assert device.is_fan
+
+    await device.set_fan_speed_level(1)
+    await dev.update()
+    assert device.fan_speed_level == 1
+    assert device.is_on
+
+    await device.set_fan_speed_level(4)
+    await dev.update()
+    assert device.fan_speed_level == 4
+
+    await device.set_fan_speed_level(0)
+    await dev.update()
+    assert not device.is_on
+
+    with pytest.raises(KasaException):
```

Comment:
These should raise a standard `ValueError`.

Reply:
Done